### PR TITLE
Add recipe for building fann (Fast Artificial Neural Network) library

### PR DESCRIPTION
**Pull request by smithsp**

### CODE REVIEW

File: recipes/fann2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  fn: fann2-2.2.0.tar.gz
+  url: https://github.com/libfann/fann/archive/2.2.0.tar.gz
+  md5:  e78dc740982c083154980348a749038c
+  sha256: f31c92c1589996f97d855939b37293478ac03d24b4e1c08ff21e0bd093449c3c
+#  patches:
+   # List any patch files here
+   # - fix.patch
+
+build:
+  number: 1
```
Comment (CJ-Wright) [inline recipes/fann2/meta.yaml]: Is this a new recipe? If so then the build number should be `0`

File: recipes/fann2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package:
+  name: fann2
+  version: "2.2.0"
+
+source:
+  fn: fann2-2.2.0.tar.gz
+  url: https://github.com/libfann/fann/archive/2.2.0.tar.gz
+  md5:  e78dc740982c083154980348a749038c
```
Comment (CJ-Wright) [inline recipes/fann2/meta.yaml]: I don't think you need both, we usually prefer sha256.

File: recipes/fann2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  build:
+    - cmake
+
+test:
+  commands:
+    - test -f $PREFIX/lib/libdoublefann.dylib  # [osx]
+    - test -f $PREFIX/lib/libfann.dylib  # [osx]
+    - test -f $PREFIX/lib/libfixedfann.dylib  # [osx]
+    - test -f $PREFIX/lib/libfloatfann.dylib  # [osx]
+    - test -f $PREFIX/lib/libdoublefann.so  # [unix]
```
Comment (CJ-Wright) [inline recipes/fann2/meta.yaml]: Are both `dylib` and `so` built on OSX?

**Comment by smithsp** [comment]
@CJ-Wright Thanks for your feedback.

**Pull request merged**